base/threadingconstructs.jl

```diff
@@ -57,10 +57,21 @@ Returns the number of threadpools currently configured.
 """
 nthreadpools() = Int(unsafe_load(cglobal(:jl_n_threadpools, Cint)))
 
+"""
+    Threads.nthreadpool()
+
+Get the number of threads available to the Julia default worker thread-pool.
+
+See also: `BLAS.get_num_threads` and `BLAS.set_num_threads` in the
+[`LinearAlgebra`](@ref man-linalg) standard library, and `nprocs()` in the
+[`Distributed`](@ref man-distributed) standard library.
+"""
+nthreadpool() = Threads._nthreads_in_pool(Int8(0))
```

Comment:
Choose a better name, to be less confusable with `nthreadpools`. Possibly `sizethreadpool`?

Reply:
`defaultpoolsize`?

Reply:
Why not
`nthreads()` -> default pool
`nthreads(pool)`

`nthreadpool` seems confusing IMO

Reply:
`nthreads()` means something different now, so that would probably violate semver

Reply:
True, `nthreads()` currently means the total number of threads over all pools, so we should just keep it that way. `nthreads(pool)` seems fine to me too. If we want a new function for the number of threads in the default pool we can maybe add that, but it doesn't need to be done in this PR.

Reply:
But after rebasing, this seems to be running into some sort of memory issue now, so I am trying to figure out what it is conflicting with